File: sdk/src/main/java/com/mrboomdev/awery/sdk/util/UniqueIdGenerator.java
package com.mrboomdev.awery.sdk.util;

/**
 * A helper-class which helps to generate random ids which doesn't collide with other ids.
 * @author MrBoomDev
 */
public class UniqueIdGenerator {
	private long usedLongs;
	private int usedIntegers;

	/**
	 * @return A new id of type Long
	 * @see #getInteger()
	 * @author MrBoomDev
	 */
	public long getLong() {
		return usedLongs++;
	}

	/**
	 * @return A new id of type Integer
	 * @see #getLong()
	 * @author MrBoomDev
	 */
	public int getInteger() {
		return usedIntegers++;
	}

	/**
	 * Clears the pool of used ids. Next call to getLong() and getInteger()
	 * may return values that do collide with those that were generated before this method call.
	 * @author MrBoomDev
	 */
	public void clear() {
		usedLongs = 0;
		usedIntegers = 0;
	}
}

File: app/src/main/java/com/mrboomdev/awery/util/UniqueIdGenerator.kt
package com.mrboomdev.awery.util

class UniqueIdGenerator {
    private var initialValue = 0L
    private var value = 0L

    constructor(initialValue: Long) {
        this.value = initialValue
        this.initialValue = initialValue
    }

    constructor(): this(0)

    val long: Long
        get() = ++value

    val integer: Int
        get() = long.toInt()

    val float: Float
        get() = long.toFloat()

    fun reset(initialValue: Long) {
        this.initialValue = initialValue
        this.value = initialValue
    }

    fun reset() {
        reset(initialValue)
    }
}